### PR TITLE
HAAR-881 Add endpoint to find emails of users stored in auth

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/config/AbstractWebClientConfiguration.kt [hunk 1]
```diff
@@ -28,7 +28,6 @@ abstract class AbstractWebClientConfiguration(appContext: ApplicationContext, pr
     .clientName(clientId)
     .clientId(environment.getRequiredProperty("$clientId.client.client-id"))
     .clientSecret(environment.getRequiredProperty("$clientId.client.client-secret"))
-    .clientSecret(environment.getRequiredProperty("$clientId.client.client-secret"))
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/config/AbstractWebClientConfiguration.kt:31]: noticed duplicate line so removed

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserController.kt [hunk 2]
```diff
@@ -316,7 +317,6 @@ class UserController(private val userService: UserService) {
     }
     .orElseGet { notFoundResponse(username) }
 
-  @Deprecated("Please use /api/authuser/email instead")
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/UserController.kt:319]: removed as not yet deprecated and association with endpoint is wrong

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/PrisonUserController.kt [hunk 3]
```diff
@@ -89,6 +93,24 @@ class PrisonUserController(
...
+    summary = "Find email addresses for prison users stored in auth",
+    description =
+    """ Finds verified and unverified email address for users stored in auth.
+        Post version that accepts multiple email addresses.
+        Requires ROLE_MAINTAIN_ACCESS_ROLES or ROLE_MAINTAIN_ACCESS_ROLES_ADMIN.
+    """
+  )
+
+  @PreAuthorize("hasAnyRole('ROLE_MAINTAIN_ACCESS_ROLES', 'ROLE_MAINTAIN_ACCESS_ROLES_ADMIN')")
+  fun getUserEmails(
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/resource/api/PrisonUserController.kt:107]: This is similar to an existing endpoint to get a list of emails in UserController but this endpoint returns the email stored in auth

**Review by AlTurner-MOJ — APPROVED** [review]
lgtm

**Pull request merged**